WolvenKit.Modkit/RED4/Tools/MeshTools.cs

```diff
@@ -550,12 +557,26 @@ public static List<RawMeshContainer> ContainRawMesh(MemoryStream gfs, MeshesInfo
                     meshContainer.indices[i] = gbr.ReadUInt16();
                 }
 
+                if (meshname.Length>0)
+                {
+                    meshContainer.name = mergeMeshes ?
+                                        info.appearances.Keys.FirstOrDefault("default") :
+                                        Path.GetFileNameWithoutExtension(meshname)+"_submesh_" + Convert.ToString(index).PadLeft(2, '0') + "_LOD_" + info.LODLvl[index];
 
-                meshContainer.name = mergeMeshes ?
-                    info.appearances.Keys.FirstOrDefault("default") :
-                    "submesh_" + Convert.ToString(index).PadLeft(2, '0') + "_LOD_" + info.LODLvl[index];
+                }
+                else
+                {
+                    meshContainer.name = mergeMeshes ?
+                                        info.appearances.Keys.FirstOrDefault("default") :
+                                        "submesh_" + Convert.ToString(index).PadLeft(2, '0') + "_LOD_" + info.LODLvl[index];
+                }
+                
```

Comment:
As far as I can tell its an arbitrary naming convention that wkit decided to use for the submeshes in an object at some point. the whatever is just the mesh index of the submeshes.

Reply:
My concern with this area is just ensuring we're being as agnostic as possible with our mesh naming conventions. Blender/glTF is what we support the most at the moment, but there's been talk about trying to bring back other file formats and I'd like to avoid writing specifically for one program/format over another.

Reply:
its no less agnostic than it was before this patch, its just more useable with multimesh.